File: leetcode/0100_same-tree/solution.c
#include <stdio.h>

struct TreeNode {
    int val;
    struct TreeNode *left;
    struct TreeNode *right;
};

/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     struct TreeNode *left;
 *     struct TreeNode *right;
 * };
 */
int isSameTree(struct TreeNode *p, struct TreeNode *q)
{
    if (p == NULL && q == NULL)
        return 1;
    if (p == NULL || q == NULL)
        return 0;
    if (p->val != q->val)
        return 0;
    return isSameTree(p->left, q->left) && isSameTree(p->right, q->right);
}

File: leetcode/0100_same-tree/solution.js
var { TreeNode, utils } = require('../dtype');

/**
 * @param {TreeNode} p
 * @param {TreeNode} q
 * @return {boolean}
 */
var isSameTree = function(p, q) {
  if (!p && !q) return true;
  if (!p || !q) return false;
  if (p.val !== q.val) return false;
  return isSameTree(p.left, q.left) && isSameTree(p.right, q.right);
};

// 测试
var tree1 = utils.tree(1, 2, 3);
var tree2 = utils.tree(1, 2, 3);
var tree3 = utils.tree(1, 2, 1);
console.log(
  isSameTree(tree1, tree2),
  isSameTree(tree1, tree3)
);
